File: src/jvis/data/stacks/vue-vite/files/src/types/item.ts
export interface Item {
  id: string;
  name: string;
  description: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateItemInput {
  name: string;
  description?: string | null;
}

export interface UpdateItemInput {
  name?: string;
  description?: string | null;
}

File: src/jvis/data/stacks/python-fastapi/files/src/domain/schemas/item.py
"""Pydantic schemas for Item API request/response validation."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ItemCreate(BaseModel):
    name: str
    description: str | None = None


class ItemUpdate(BaseModel):
    name: str | None = None
    description: str | None = None


class ItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    description: str | None
    created_at: datetime
    updated_at: datetime


class ItemList(BaseModel):
    items: list[ItemResponse]
    count: int
